File: IRSClientAngular/src/app/component/location/campus.ts
import { Config } from '../../util/config.service';
export class Campus {
    table: string;
    attributes: CampusAttributes;

    constructor() {
        this.table = Config.CampusTable;
        this.attributes = new CampusAttributes();
    }

    toSearchString(): string {
        return this.attributes.CITY 
        + " " + this.attributes.ADDRESS 
    };
}

export class CampusAttributes {
    CAMPUS_ID: number;
    CITY: string;
    ADDRESS: string;
}

File: WebApiSpark/src/main/java/Model/Campus.java
package Model;


import Util.DatabaseValues;

public class Campus extends IncidentElement
{
    public Campus ( )
    {
        this (
                "",
                "",
                ""
        );
    }

    public Campus (
            String campusID,
            String city,
            String address
    ) {
        super (
                DatabaseValues.Table.CAMPUS,
                new DatabaseValues.Column[]
                        {
                                DatabaseValues.Column.CAMPUS_ID,
                                DatabaseValues.Column.CITY,
                                DatabaseValues.Column.ADDRESS,
                        }
        );

        updateAttributeValue(
                DatabaseValues.Column.CAMPUS_ID,
                campusID
        );

        updateAttributeValue(
                DatabaseValues.Column.CITY,
                city
        );

        updateAttributeValue(
                DatabaseValues.Column.ADDRESS,
                address
        );
    }

}
